### ISSUE TITLE
Machine Learning / Resampling Methods: suggested clarifications

### ISSUE THREAD

**Issue by hbmaclean** [rafalab/dsbook-part-2#40]
[Machine Learning/Resampling Methods](https://rafalab.dfci.harvard.edu/dsbook-part-2/ml/resampling-methods.html#parameter-tuning) has this:

> > First, note that the estimate obtained on the training set is generally lower than the estimate obtained with the test set, with the difference larger for smaller values of . This is due to over-training.
> >
Might it also be due to the slightly different distributions of 7s and 2s in the two data sets?  What I notice is a near-equal difference between the distribution ratios and accuracies (both off about 2%).

> > So do we simply pick the  that maximizes accuracy and report this accuracy? There are two problems with this approach:
> >
I think you need to clarify which set, presumably the training value - i.e., pick the optimal from training, freeze it, then test with it.

> > The accuracy versus  plot is quite jagged. We do not expect this because small changes in  should not affect the algorithm’s performance so much. The jaggedness is explained by the fact that the accuracy is computed on a sample and therefore is a random variable.
> >
It sure is jagged... and the y-axis does not start at 0.  I realize starting with zero is not a hard-and-fast rule, but doing so here makes the differences appear rather insignificant.  How do you balance those concerns - showing the difference, while not exaggerating?

> > Although for each  x we estimated MSE using the test set, we used the test set to pick the best . As a result, we should not expect this minimum test set accuracy to extrapolate to the real world.

The first sentence is unclear.  Starting with "although" and then mentioning the same dataset in both phrases is conflicting.
I *think* what I've learned is that we:

1. Grab optimal estimate from training
2. Use it against the test set
3. Have the resulting test-set confidence that the training value will extrapolate to the real world
4. Or, reiterate 1-3 until confident in confidence
